source/GameInterface/Services/Settlements/Patches/SetWallHitPointsSettlementPatch.cs

```diff
@@ -0,0 +1,49 @@
+﻿using Common;
+using Common.Messaging;
+using Common.Util;
+using GameInterface.Extentions;
+using GameInterface.Policies;
+using GameInterface.Services.Settlements.Messages;
+using HarmonyLib;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using TaleWorlds.CampaignSystem.Settlements;
+using TaleWorlds.Library;
+
+namespace GameInterface.Services.Settlements.Patches;
+
+/// <summary>
+/// Used to Patch Settlement.SetWallSectionHitPointsRatioAtIndex() server side sync.
+/// </summary>
+[HarmonyPatch(typeof(Settlement))]
+internal class SetWallHitPointsSettlementPatch
+{
+    [HarmonyPatch("SetWallSectionHitPointsRatioAtIndex")]
+    [HarmonyPrefix]
+    private static bool SetWallSectionHitPointsRatioAtIndexPrefix(ref Settlement __instance, ref int index, ref float hitPointsRatio)
+    {
+        if (AllowedThread.IsThisThreadAllowed()) return true;
+        if (CallOriginalPolicy.IsOriginalAllowed()) return true;
+        if (ModInformation.IsClient) return false;
```

Comment:
Change this to match up to line 42 of https://github.com/Bannerlord-Coop-Team/BannerlordCoop/blob/development/source/GameInterface/Services/Armies/Patches/ArmyLifetimePatches.cs#L42